docs/src/pages/components/data-grid/export/export.md

```diff
@@ -57,6 +57,21 @@ There are a few ways to include or hide other columns:
 <DataGrid columns={[{ field: 'id', disableExport: true }, { field: 'brand' }]} />
 ```
 
+#### Exported rows
+
+By default, if some rows are selected, the CSV export will only contain those rows.
+If not, the CSV export will contain all the rows passing the filtering process and not collapsed.
+
+Set the `getRowsToExport` to manually pick the rows to include in the CSV export.
+
+{{"demo": "pages/components/data-grid/export/CsvGetRowsToExport.js", "bg": "inline", "defaultCodeOpen": false}}
+
+When using [Row grouping](/components/data-grid/group-pivot/#row-grouping), it can be useful to remove the groups from the CSV export
+
+**Note**: For now, it is not possible to export the collapsed row without first expanding them.
```

Comment:
After reading https://github.com/mui-org/material-ui-x/issues/3610#issuecomment-1014584249 I was under impression that `getRowsToExport` will export collapsed rows as well.
Have you changed your mind about this?

Reply:
I have a technical limitation that make it hard to get the collapsed rows post-filter
So I'm introducing the callback but not changing the default behavior for now.
I will try to change to overall filtering strategy later to enable this behavior by default, or at least add an example on this page.

Reply:
Got it, thanks for clarification!